src/Archives/AdaptiveHuffmanTree.cpp

```diff
@@ -165,36 +171,28 @@ namespace Archives
 
 
 
-	// **NOTE**: The following procedure is experimental and is not yet in use.
-	//			Nor should it be used yet!
-	/*
 	// Used during compression to get the bitstring to emit for a given code.
-	// The number of bits in the bitstring is returned and the bitstring is placed
-	// in the bitString parameter. The branch to take between the root and a child
-	// of the root is placed in the LSB. Subsequent branches are stored in higher bits
-	// **NOTE**: I may change the bit ordering! (Make that I WILL change the bit ordering)
-	int AdaptiveHuffmanTree::GetEncodedBitString(int code, int &bitString)
+	// Returns the bitstring for a given code
+	// Places the length of the bitstring in the bitCount out parameter
+	// NOTE: Bit order subject to change (and likely will change). Currently:
+	// The branch to take between the root and a child of the root is placed in the LSB
+	// Subsequent branches are stored in higher bits
+	int AdaptiveHuffmanTree::GetEncodedBitString(int code, unsigned int& bitCount)
 	{
-		// Make sure the code is in range
-		if (code >= terminalNodeCount)
-		{
-			throw std::runtime_error("Code value is out of range");
-		}
-
-		// Get the node containing the given code
-		NodeIndex curNodeIndex = parentIndex[code];
+		VerifyValidDataValue(code);
 
 		// Record the path to the root
-		bitString = 0;
-		int bitCount = 0;
+		bitCount = 0;
+		int bitString = 0;
```

Comment:
An int called bitString seems a bit strange to me. Not sure if there is a better name? It may just be because I'm not familiar with how Huffman trees work though.

Reply:
Hmm, do you feel there is confusion with the `string` data type? The `bitString` is supposed to represent a variable number of bits, which are generally processed one at a time. You might think of it as a vector of `bool`, however, you can't ask the compiler to produce an array or vector of `bool` and have it pack it for you, hence an integer base type is normally used.

In this context, the expected number of bits is about 9, though as the tree rebalances, the length can skew both shorter and longer. I expect the codes to be less than 32-bits, hence why I just used an `int`. Admittedly an `unsigned` integer type would be better. I am playing a bit fast and loose with the type here.

As for the maximum code length, it is actually as long as the number of codes. Such a length would correspond to a maximum unbalanced tree in a linear configuration. However, to unbalance the tree to a point where you have greater than 32-bit code lengths, I suspect you'd have to process about 4GB+ of specially crafted data.

I was considering building a `BitString` class, but then found myself wondering if it is sufficiently different from the `BitStream` class. Maybe I can just pass a `BitStream` class to be used by the tree code. In particular, I found myself replicating tree traversal algorithms in the test code, so perhaps that can be wrapped into tree methods.

At any rate, dealing with the type in a more precise way seems like its own branch. I can't think of a better name for what the value represents, but the type can certainly be improved.

Reply:
Thanks for explaining. I think bitString sounds good in this context.